app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -140,9 +169,7 @@ public void onNFCButtonMainActivityClick(final View view) {
      */
     public void onQRButtonMainActivityClick(final View view) {
         LOGGER.info("QR button clicked");
-        final Intent intent = new Intent(this, QRActivity.class);
-        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
-        this.startActivity(intent);
+        pairingButtonClicked(QRActivity.class);
```

Comment:
Maybe you can use an approach similar to what I used [here](https://github.com/ericcornelissen/NervousFish/blob/app/select-verification-activity%23183/app/src/main/java/com/nervousfish/nervousfish/activities/SelectVerificationMethodActivity.java#L69) to solve this problem?